supabase/functions/_shared/hmac-validation.ts

```
@@ -247,3 +247,179 @@ export function createWebhookValidator(secret: string, strictMode = false) {
+ * for details. This is intentionally log-only for the shadow-mode rollout.
+ *
+ * Never blocks — shadow mode only.
+ */
+/**
+ * gmail-webhook e publico e este token nao passou por verificacao de
+ * assinatura: qualquer um pode escolher aud/iss. Sem sanitizar, um CR/LF no
+ * claim forja linhas `[WEBHOOK_AUTH_SHADOW]` inteiras no log.
+ */
+function sanitizeClaimForLog(value: unknown): string {
+  if (typeof value !== 'string') return value === undefined ? '(ausente)' : '(invalido)';
+  const flat = value.replace(/[\u0000-\u001f\u007f]/g, ' ');
```

ESLint

[error] Unexpected control character(s) in regular expression: \x00, \x1f.

(no-control-regex)